File: XWeather/Shared/Domain/Wu/Forecast.cs
using System.Collections.Generic;

namespace XWeather.Domain
{
	public class Forecast : WuObject
	{
		public override string WuKey => "forecast";

		public ForecastDetail forecast { get; set; }

		public TxtForecast TextForecast => forecast?.txt_forecast;
		public SimpleForecast Simpleforecast => forecast?.simpleforecast;
	}


	public class ForecastDetail
	{
		public TxtForecast txt_forecast { get; set; }
		public SimpleForecast simpleforecast { get; set; }
	}


	public class TxtForecastDay
	{
		public int period { get; set; }
		public string icon { get; set; }
		public string icon_url { get; set; }
		public string title { get; set; }
		public string fcttext { get; set; }
		public string fcttext_metric { get; set; }
		public string pop { get; set; }
	}


	public class ForecastDay
	{
		public WuDate date { get; set; }
		public int period { get; set; }
		public Temperature high { get; set; }
		public Temperature low { get; set; }
		public string conditions { get; set; }
		public string icon { get; set; }
		public string icon_url { get; set; }
		public string skyicon { get; set; }
		public int pop { get; set; }
		public Precipitation qpf_allday { get; set; }
		public Precipitation qpf_day { get; set; }
		public Precipitation qpf_night { get; set; }
		public Precipitation snow_allday { get; set; }
		public Precipitation snow_day { get; set; }
		public Precipitation snow_night { get; set; }
		public Wind maxwind { get; set; }
		public Wind avewind { get; set; }
		public int avehumidity { get; set; }
		public int maxhumidity { get; set; }
		public int minhumidity { get; set; }
	}


	public class TxtForecast
	{
		public string date { get; set; }
		public List<TxtForecastDay> forecastday { get; set; }
	}


	public class SimpleForecast
	{
		public List<ForecastDay> forecastday { get; set; }
	}
}

File: XWeather-Swift/XWeather/Domain/Forecast.swift
//
//  Forecast.swift
//  XWeather
//
//  Created by Colby Williams on 6/11/17.
//  Copyright © 2017 Colby Williams. All rights reserved.
//

import Foundation

public class Forecast : WuObject
{
	public override var WuKey: String { get { return "forecast" } }
	
	public var forecast: ForecastDetail?
	
	public var textForecast: TxtForecast?
	public var simpleforecast: SimpleForecast?
}


public class ForecastDetail
{
	public var txt_forecast: TxtForecast?
	public var simpleforecast: SimpleForecast?
}


public class TxtForecastDay
{
	public var period: Int?
	public var icon: String?
	public var icon_url: String?
	public var title: String?
	public var fcttext: String?
	public var fcttext_metric: String?
	public var pop: String?
}


public class ForecastDay
{
	public var date: WuDate?
	public var period: Int?
	public var high: Temperature?
	public var low: Temperature?
	public var conditions: String?
	public var icon: String?
	public var icon_url: String?
	public var skyicon: String?
	public var pop: Int?
	public var qpf_allday: Precipitation?
	public var qpf_day: Precipitation?
	public var qpf_night: Precipitation?
	public var snow_allday: Precipitation?
	public var snow_day: Precipitation?
	public var snow_night: Precipitation?
	public var maxwind: Wind?
	public var avewind: Wind?
	public var avehumidity: Int?
	public var maxhumidity: Int?
	public var minhumidity: Int?
}


public class TxtForecast
{
	public var date: String?
	public var forecastday: [TxtForecastDay] = []
}


public class SimpleForecast
{
	public var forecastday: [ForecastDay] = []
}
